### PR TITLE
fix: Enabled domain discovery when external_dns is disabled

### LINKED ISSUE #24: Ingress IgnoreLoadBalancer property needs to be set to false
Based on https://github.com/jenkins-x/jx/issues/7537 the Ingress ignoreLoadBalancer property needs to be set to false in the `jx-requirements.yml.tpl` file.

> So, after this PR #7515, we need to set Ingress.IgnoreLoadBalancer in the requirements file to false.

This configuration seems to fix issues standing up the vault:

```
ingress:
  domain: "tfjxevidentmarlin"
  ignoreLoadBalancer: false
  externalDNS: false
  tls:
    email: ""
    enabled: true
    production: true
```

### PR DESCRIPTION
**Pull request by chrismellard**
Resolves #24

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //
 // ----------------------------------------------------------------------------
 terraform {
-  required_version = ">= 0.12.17"
+  required_version = ">= 0.13.0"
```
Comment (ankitm123) [inline main.tf:6]: Using features only in 0.13?

Comment (chrismellard) [reply main.tf:6]: No. Just given the CI is running against 0.13 now this feels the safest approach unless I run two CI stages for 0.12 and 0.13. 0.13 is in line with the latest jx builders. Thoughts?

Comment (ankitm123) [reply main.tf:6]: Feels a bit too early to stop supporting 0.12 imo. For eks, I think I will go with supporting 0.12 as long as I dont use a feature which is exclusive to 0.13. I see most modules supporting both versions, so ...
Two CI builds (matrix build of some sort) would be ideal, but there are resource constraints.

Comment (ankitm123) [reply main.tf:6]: It's fine to test with 0.13 only, but we should not force the users to upgrade to 0.13 yet. wdyt?

Comment (chrismellard) [reply main.tf:6]: Can go with 0.12 and CI both versions. Though the conditional validation on TLS variables in #22 might make this a moot point - I think that is using 0.13 only functionality. Will wait and see what that potential PR looks like first before I commit  either way

Comment (chrismellard) [reply main.tf:6]: Ok. See issue #22. I quite like the conditional validation here - so long as the variables are related and warrant residing within the same object I think this keeps cognitive load quite low. So we can either merge this one with 0.13 or merge the PR I'm about to open for #22 with 0.13. Your call :)

Comment (ankitm123) [reply main.tf:6]: I guess it's fine then. You can go with 0.13 (as it will be needed for #22) :+1:

**Pull request merged**